lib/tty/command/process_runner.rb

```diff
@@ -169,6 +166,15 @@ def read_stream(stream, buffer)
         end
       end
 
+      def next_written_line(stream)
```

Comment:
As far as I can tell this implementation 'accidentally' solves the problem. If an external process doesn't produce any output the `gets` won't hung as it won't be able to consume anything and a thread will die immediately. However, due to timeout setup it will raise an error and hence stop the script.